src/components/FormComponent/GuestForm.jsx

```diff
@@ -1,12 +1,16 @@
 import { useState } from 'react';
+import { useHistory } from 'react-router';
 import { useEntries } from '../../context/EntryContext';
 import { useUser } from '../../context/UserContext';
+import { useAuth } from '../../hooks/useAuth';
 
 export default function GuestForm() {
   const [name, setName] = useState('');
   const [guestEntry, setGuestEntry] = useState('');
   const { user, setUser } = useUser();
   const { entries, setEntries } = useEntries();
+  const history = useHistory();
+  const { logout } = useAuth();
 
   function updateGuestName() {
```

Comment:
This function isn't necessary anymore since now you should be referencing the username/email that is stored in auth context from the login credentials. Instead of this process, you wouldn't need to manage the user's name in local state, but instead can refer to it from the user you get from your global context. Then you won't need to clear the name either in the logout function.